### PR TITLE
ENT-9402: Added policy to facilitate migration of ignore_interfaces.rx from inputdir to workdir (3.21.x)

### PR DESCRIPTION
**Pull request by larsewi**

### CODE REVIEW

File: cfe_internal/recommendations.cf [hunk 1]
```diff
@@ -9,6 +9,66 @@ bundle agent MPF_class_recommendations
...
+
+  files:
+    default:mpf_auto_migrate_ignore_interfaces_rx_to_workdir.ignore_interfaces_in_inputdir::
+      "$(sys.workdir)/ignore_interfaces.rx" -> { "ENT-9402" }
+        copy_from => local_dcp( "$(sys.inputdir)/ignore_interfaces.rx"),
+        comment => concat( "Excluding interfaces should be done outside of the",
+                           " policy input directory so that it's easier to",
+                           " ignore different interfaces on different hosts.");
+
+    default:cfengine_3::
```
Comment (craigcomstock) [inline cfe_internal/recommendations.cf:45]: why cfengine_3 here? Is ignore_interfaces.rx only in version >=3? Not that we do much with version <3.

File: MPF.md [hunk 2]
```diff
@@ -597,6 +597,47 @@ This [augments file][Augments] is a way to specify that `cf-monitord` should be
 The following settings are defined in `controls/def.cf` can be set from an
 [augments file][Augments].
 
+### Automatically migrate ignore_interfaces.rx to workdir
+
+`ignore_interfaces.rx` defines regular expressions matching network interfaces that CFEngine should ignore.
+
+Prior to `3.23.0` this file was expected to be found in
+`$(sys.inputdir)/ignore_interfaces.rx`. Beginning with `3.23.0` preference is
+given to `$(sys.workdir)/ignore_interfaces.rx` if it is found and `WARNING` is
```
[the remark this answers is not in the document]
Comment (craigcomstock) [reply MPF.md:606]: Ah, seeing the policy below I see that it is both a `WARNING` and a report. Maybe mention the report here as well?

File: MPF.md [hunk 3]
```diff
@@ -597,6 +597,47 @@ This [augments file][Augments] is a way to specify that `cf-monitord` should be
...
+      "comment": "Automatically migrate ignore_interfaces.rx to workdir."
+    }
+  }
+}
+```
+
+Additionally, to disable reports about the presence of
+`$(sys.inputdir)/ignore_interfaces.rx` define the class
+`default:mpf_auto_migrate_ignore_interfaces_rx_to_workdir_reports_disabled`.
+When this class is not defined, `cf-agent` will emit reports indicating it's
```
[the remark this answers is not in the document]
Comment (craigcomstock) [reply MPF.md:627]: I see both are generated. OK. :)

File: cfe_internal/recommendations.cf [hunk 4]
```diff
@@ -9,6 +9,66 @@ bundle agent MPF_class_recommendations
...
+# @brief Recommend completing migration of ignore_interfaces.rx from inputdir to workdir
+#
+# @description This bundle identifies the presence of ignore_interfaces.rx in
+# inputdir and emits messages about its presence. Additionally, the file is
+# automatically copied if the file is present in inputdir and the class
+# default:mpf_auto_migrate_ignore_interfaces_rx_to_workdir is defined (not
+# defined by default).
+{
+  meta:
+      "tags" slist => { "cfengine_recommends" };
```
[the remark this answers is not in the document]
Comment (nickanderson) [reply cfe_internal/recommendations.cf:28]: While testing last night I found that it wasn't working, it was searching for the wrong  tag. So, it should have been used but was not.

**Comment by nickanderson** [comment]
@larsewi  I have a pr doing a little refactor here to prep for backport.
https://github.com/cfengine/masterfiles/pull/2766

**Comment by nickanderson** [comment]
Superseded by https://github.com/cfengine/masterfiles/pull/2768

**Pull request closed without merging**